client/blocks/login/index.jsx

```diff
@@ -45,6 +44,27 @@ import ErrorNotice from './error-notice';
 import LoginForm from './login-form';
 import './style.scss';
 
+/*
+ * Parses the `anchor_podcast` parameter from a given URL.
+ * Returns `true` if provided URL is an anchor FM signup URL.
+ */
+function getIsAnchorFmSignup( urlString ) {
+	if ( ! urlString ) {
+		return false;
+	}
+
+	// Assemble search params if there is actually a query in the string.
+	const queryParamIndex = urlString.indexOf( '?' );
+	if ( queryParamIndex === -1 ) {
+		return false;
+	}
+	const searchParams = new URLSearchParams(
+		decodeURIComponent( urlString.slice( queryParamIndex ) )
+	);
+	const anchorFmPodcastId = searchParams.get( 'anchor_podcast' );
+	return Boolean( anchorFmPodcastId && anchorFmPodcastId.match( /^[0-9a-f]{7,8}$/i ) );
+}
+
```

Comment:
We have another one of these [here](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding/client/landing/stepper/declarative-flow/anchor-fm-flow.ts#L14). Maybe worth putting in the `onboarding` package.

Reply:
Given that those are different functions, I preferred not to introduce additional dependencies to the Login block used in multiple places, but this is definitely something worth arranging at some point.